i6_models/config.py

```diff
@@ -29,7 +29,18 @@ class ModelConfiguration:
     def _validate_types(self) -> None:
         for field in fields(type(self)):
             try:
-                typeguard.check_type(getattr(self, field.name), field.type)
+                if isinstance(field.type, str):
```

Comment:
You should add some comment. Maybe referring to the issue you should open on the typeguard repo about how to handle this.